examples/coco_example.py

```diff
@@ -0,0 +1,446 @@
+                                 _MODEL_TARBALL_URL)
+
+  print("Original model files at {}".format(tarball))
+  with tarfile.open(tarball) as t:
+    frozen_graph_bytes = t.extractfile(_FROZEN_GRAPH_MEMBER).read()
+    return tf.GraphDef.FromString(frozen_graph_bytes)
+
+
+def _build_preprocessing_graph_def():
+  """
+  Build a TensorFlow graph that performs the preprocessing operations that
+  need to happen before the main graph, then convert to a GraphDef.
+
+  Returns:
+    Python object representation of the GraphDef for the preprocessing graph.
+    Input node of the graph is the placeholder "raw_image", and the output is
+    the node with the name "preprocessed_image".
+  """
+  # At the moment, the only preprocessing we need to perform is converting
+  # JPEG/PNG/GIF files to numpy arrays.
+  img_decode_g = tf.Graph()
+  with img_decode_g.as_default():
+    raw_image = tf.placeholder(tf.string, name="raw_image")
+
+    # Downstream code hardcodes RGB
+    _NUM_CHANNELS = 3
+
+    # The TensorFlow authors, in their infinite wisdom, created two generic
+    # image-decoder ops. tf.image.decode_imaage() returns a 4D tensor when it
+    # receives a GIF and a 3D tensor for every other file type. This means
+    # that you need complicated shape-checking and reshaping logic downstream
+    # for it to be of any use in an inference context.
+    # The other op is tf.image.decode_png(). In spite of its name, this op
+    # actually handles PNG, JPEG, and non-animated GIF files. For now, we use
+    # this op for simplicity.
+    decoded_image = tf.image.decode_png(raw_image, _NUM_CHANNELS)
+
+    # Downstream code expects a batch of equal-sized images. For now, we
+    # generate a single-image batch.
+    decoded_image_batch = tf.expand_dims(decoded_image, 0,
+                                         name="preprocessed_image")
+
+  return img_decode_g.as_graph_def()
+
+
+def _build_postprocessing_graph_def():
+  """
+  Build the TensorFlow graph that performs postprocessing operations that
+  should happen after the main graph.
+
+  Returns:
+    Python object representation of the GraphDef for the postprocessing graph.
+    The graph has one input placeholder called "detection_classes" and
+    an output op called "decoded_detection_classes".
+    The graph will also have an op called "hash_table_init" that initializes
+    the mapping table. This op MUST be run exactly once before the
+    "decoded_detection_classes" op will work.
+  """
+  label_file = _fetch_or_use_cached("labels.pbtext", _LABEL_MAP_URL)
+
+  # Category mapping comes in pbtext format. Translate to the format that
+  # TensorFlow's hash table initializers expect (key and value tensors).
+  with open(label_file, "r") as f:
+    raw_data = f.read()
+  # Parse directly instead of going through the protobuf API dance.
+  records = raw_data.split("}")
+  records = records[0:-1]  # Remove empty record at end
+  records = [r.replace("\n", "") for r in records] # Strip newlines
```

Comment:
nit: 2 spaces before comment

Reply:
Thanks, not sure why IntelliJ didn't flag that.